nixos/modules/security/acme.nix

```diff
@@ -399,7 +399,17 @@ in
       systemd.tmpfiles.rules =
         map (data: "d ${data.webroot}/.well-known/acme-challenge - ${data.user} ${data.group}") (filter (data: data.webroot != null) (attrValues cfg.certs));
 
-      systemd.timers = flip mapAttrs' cfg.certs (cert: data: nameValuePair
+      systemd.timers = let
+        # Allow systemd to pick a convenient time within the day
+        # to run the check.
+        # This allows the coalescing of multiple timer jobs.
+        # We divide by the number of certificates so that if you
+        # have many certificates, the renewals are distributed over
+        # the course of the day to avoid rate limits.
+        numCerts = length (attrNames cfg.certs);
+        _24hSecs = 60 * 60 * 24;
+        AccuracySec = if numCerts == 0 then null else "${toString (builtins.div _24hSecs numCerts)}s";
```

Comment:
It's early here and I'm fuzzy so pardon if I'm completely misreading... but won't `numCerts` always be non zero?

Reply:
Correct, as we asser that `cfg.certs != {}` at the top. though that's a bit of a weird check in my opinion

Reply:
Makes sense; I was just defensively avoiding the division by zero, though I suppose with lazy evaluation there's not really much point. I can adjust this if you'd prefer, though I'm not entirely convinced that that assert will last forever :)